Makefile

```diff
@@ -561,30 +570,32 @@ COVER_PKGS = client common host service tools
 # pkg -> pkg/... -> github.com/uber/cadence/pkg/... -> join with commas
 GOCOVERPKG_ARG := -coverpkg="$(subst $(SPACE),$(COMMA),$(addprefix $(PROJECT_ROOT)/,$(addsuffix /...,$(COVER_PKGS))))"
 
+# iterates over a list of dirs and runs go test on each one, collecting errors as it runs.
+# this is primarily written because it's a verbose bit of boilerplate, until we switch to `go test ./...` where possible.
+# CAUTION: when changing to `go test ./...`, note that this DOES NOT test submodules.  Those must be run separately.
+define looptest
+$Q FAIL=""; for dir in $1; do \
+	go test $(TEST_ARG) -coverprofile=$@ "$$dir" $(TEST_TAG) | tee -a test.log || FAIL="$$FAIL $$dir"; \
```

Comment:
I think the tee needs to grab stderr too?

Reply:
That is probably true, yeah. Good catch.

Why it exists: outside human use (which I doubt anyone does on the team), it can in principle give you a simple upload to buildkite, like for coverage, and I think I saw that at one point (or maybe I'm getting it confused with internal systems).  Buildkite's "download the raw log file" format is pretty awkward comparatively.

I do mostly lean towards "not worth keeping" though. Feels like just unused legacy to me. I never did finish rewriting these makefiles, you can still see the separator where I stopped a bit above here...

Reply:
all I'm seeing it used for is build failures and probably race crashes... but yes, that seems unambiguously worth gathering into the same log file, otherwise it just says "fail: package".

added!